server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSLexer.g4

```diff
@@ -837,6 +837,7 @@ NUMERICLITERAL : (PLUSCHAR | MINUSCHAR)? DIGIT* (DOT | COMMACHAR) DIGIT+ (('e' |
 NONNUMERICLITERAL : UNTRMSTRINGLITERAL | STRINGLITERAL | DBCSLITERAL | HEXNUMBER | NULLTERMINATED;
 
 IDENTIFIER : ([a-zA-Z0-9]+ ([-_]+ [a-zA-Z0-9]+)*);
+INVALID_IDENTIFIER : [a-zA-Z0-9#@$]+ ([-_]+ [a-zA-Z0-9#@$]+)*;
```

Comment:
Is this intended?

Reply:
Yes, it is to catch the identifiers that contain `#@$` and there missed by the preprocessor. 
The logic here is this: copybook names may contain `#@$` as a part of a valid name when identifiers may not. With the previous implementation, e.g. variable name `VARNAME$` was treated as a possible copybook name by the preprocessor, and as `VARNAME` and `$` (two different tokens) by the main lexer. In this case, the mapper could not map the token one to one and the mapping might miss some tokens.
With this change, all the identifiers, even incorrect, will be mapped correctly.